File: java/Encapsulation.java
public class Encapsulation { 
  private String name;
  private String occupation; // private = restricted access

  // Getter
  public String getName() {
    return name;
  }

  // Setter
  public void setName(String newName) {
    this.name = newName;
  }

  public String getOccupation(){
    return occupation;
  }

  public void setOccupation(String newOccupation){
    this.occupation = newOccupation;
  }

  public static void main(String[] args) {
    Encapsulation myObj = new Encapsulation();
    myObj.setName("John");
    myObj.setOccupation("Developer"); 
    System.out.println("my name is " + myObj.getName() + ", my occupation is a " + myObj.getOccupation());
  }
}

File: dart_application_1/bin/encapsulation.dart
class encapsulation {
  String? _name;
  String? _occupation;

//Getter
String get name  {
  return _name!;
}

String get occupation  {
  return _occupation!;
}

//Setter
void setName (String name) {
  this._name = name;
}
void setOccupation (String occupation) {
  this._occupation = occupation;
}

}
void main () {
  encapsulation person = new encapsulation();
  person.setName("John");
  person.setOccupation("Developer");
  print("my name is ${person.name}, my occupation is a ${person.occupation}");
}